File: thccb-frontend/src/types/chart.ts
// 图表相关类型定义

/** 前端支持的 candle interval —— 跟后端 OutcomeCandle.interval CheckConstraint 对齐。
 * 30s/5m/1d 是后端 rollup-compatible，但前端不暴露主动发送。 */
export type ChartInterval = '10s' | '1m' | '15m' | '1h'

export interface Candle {
  t: string
  o: number
  h: number
  l: number
  c: number
  v: number
  n: number
}

export interface CandleSeriesResponse {
  outcome_id: number
  interval: ChartInterval
  from_ts: string
  to_ts: string
  candles: Candle[]
}

File: backend/app/schemas/chart.py
# app/schemas/chart.py

from datetime import datetime
from typing import Literal, Optional, List
from pydantic import BaseModel, Field


class PricePoint(BaseModel):
    ts: datetime
    price: float


class Candle(BaseModel):
    t: datetime  # bucket start
    o: float
    h: float
    l: float
    c: float
    v: float = 0.0         # volume (shares)
    n: int = 0             # number of trades


Interval = Literal["10s", "30s", "1m", "5m", "15m", "1h", "1d"]


class PriceSeriesResponse(BaseModel):
    outcome_id: int
    from_ts: datetime
    to_ts: datetime
    points: List[PricePoint]


class CandleSeriesResponse(BaseModel):
    outcome_id: int
    interval: Interval
    from_ts: datetime
    to_ts: datetime
    candles: List[Candle]
